artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQSession.java

```diff
@@ -397,6 +380,55 @@ public MessageProducer createProducer(final Destination destination) throws JMSE
       }
    }
 
+   void checkDestination(ActiveMQDestination destination) throws JMSException {
+      SimpleString address = destination.getSimpleAddress();
+      // TODO: What to do with FQQN
+      if (!connection.containsKnownDestination(address)) {
+         try {
```

Comment:
i will pack this slow path in a separate method to make the code more readable (perf are helped as well, but is not relevant)!

Reply:
future enhancement maybe?

Reply:
Don't know: the higher level method is already packing everything in a separate method: why wait another PR just to do the same?